docs/src/main/paradox/jms.md

```diff
@@ -305,7 +309,66 @@ Scala
 Java
 : @@snip ($alpakka$/jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java) { #assert-only-odd-messages-received }
 
-## Using Topic with an JMS provider
+### Receiving and explicitly acknowledging @extref[javax.jms.Message](javaee-api:javax.jms.Message)s from a JMS provider
+
+Create a @extref[javax.jms.Message](javaee-api:javax.jms.Message) source:
+
+Scala
+: @@snip ($alpakka$/jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsBufferedAckConnectorsSpec.scala) { #create-jms-source }
+
+Java
+: @@snip ($alpakka$/jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsBufferedAckConnectorsTest.java) { #create-jms-source }
+
+The `sessionCount` parameter controls the number of JMS sessions to run in parallel.
+
+The `bufferSize` parameter controls the maximum number of messages each JMS session will prefetch and awaiting acknowledgement before applying backpressure.
+
+Run the source and specify the amount of messages to take:
+
+Scala
+: @@snip ($alpakka$/jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsBufferedAckConnectorsSpec.scala) { #run-jms-source }
+
+Java
+: @@snip ($alpakka$/jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsBufferedAckConnectorsTest.java) { #run-jms-source }
+
+**Notes:**
+
+*  Using multiple sessions increases throughput, especially if a acknowledging message by message is desired.
+*  Messages will arrive out of order if `sessionCount` is larger than 1.
+*  Message-by-message acknowledgement can be achieved by setting `bufferSize` to 0, thus disabling buffering. The outstanding messages before backpressure will be the `sessionCount`.
+*  The default `AcknowledgeMode` is `ClientAcknowledge` but can be overridden to custom `AcknowledgeMode`s, even implementation-specific ones by setting the `AcknowledgeMode` in the `JmsSourceSettings` when creating the stream. 
```

Comment:
Isn't the default `AutoAcknowledge`?

Reply:
No. I've changed the acknowledgeMode field in the settings to be `Option[AcknowledgeMode]` allowing each source type to define their own default. The default for `ackSource` is `ClientAcknowledge`. The default for the standard no-ack mode is `AutoAcknowledge`, and the default for `txSource` is `SessionTransacted`. We still allow overriding this as seen in one test case of `ackSource` with buffering. To really prevent message loss on abort, you need to use an ActiveMQ-specific mode called `INDIVIDUAL_ACKNOWLEDGE`.